.github/workflows/build_and_test.yml

```diff
@@ -0,0 +1,44 @@
+name: Build & Test
+
+on: [push, pull_request]
+
+env: 
+  # Customize the CMake build type here (Release, Debug, RelWithDebInfo, etc.)
+  BUILD_TYPE: Release
+
+jobs:
+  generate-matrix:
+    name: Parse Job Matrix
+    runs-on: ubuntu-20.04
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Parse Job Matrix
+        id: set-matrix
+        # Note! The json in this file MUST be in a single line for parsing to succeed.
+        # Do not format matrix.json.
+        run: echo "::set-output name=matrix::$(cat ./.github/matrix.json)"
```

Comment:
That `matrix.json` file is illegible to humans. Does it need to be a single line? Maybe [jq](https://stedolan.github.io/jq/) can help? Is this a convention you've seen elsewhere?